modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -779,13 +805,28 @@ private ObjectDefinition _updateObjectDefinition(
 		objectDefinition.setPluralLabelMap(pluralLabelMap);
 
 		if (objectDefinition.isApproved()) {
+			List<ObjectEntry> objectEntries =
+				_objectEntryLocalService.getObjectEntries(
+					0, objectDefinition.getObjectDefinitionId(),
+					QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+
 			if (!active && originalActive) {
+				_deleteObjectEntriesWorkflowTasks(
+					objectDefinition, objectEntries);
+
 				objectDefinitionLocalService.undeployObjectDefinition(
 					objectDefinition);
 			}
 			else if (active) {
 				objectDefinitionLocalService.deployObjectDefinition(
 					objectDefinition);
+
+				for (ObjectEntry objectEntry : objectEntries) {
+					_objectEntryLocalService.updateObjectEntry(
```

Comment:
I think we should create a method to just call WorkflowHandlerRegistryUtil.startWorkflowInstance, what do you think?